packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -37,6 +118,17 @@ export function createAuthMiddleware({
       if (!grant || !grant.active) {
         ctx.throw(401, 'Invalid Token')
       }
+      if (!config.skipSignatureVerification) {
+        try {
+          const clientKeysService = await ctx.container.use('clientKeysService')
+          const clientKeys = await clientKeysService.getKeyByClientId(
+            grant.clientId
+          )
```

Comment:
It looks like the token introspection response will include the client's `key`
https://datatracker.ietf.org/doc/html/draft-ietf-gnap-resource-servers#section-3.3
https://github.com/interledger/rafiki/blob/3d30ce755b4218a777ec77f408be5d15dfa0e687/packages/auth/src/accessToken/routes.ts#L63
You'll need to update `authService.introspect` to also return the `key` from the response.
You could add `key` to the `Grant` class, but that'd require changing a lot of files (and conflicting with #647 ) and I don't think we actually want to be passing around the key everywhere.
:thinking: Maybe just add a:
```
interface TokenInfo extends GrantOptions {
  key: KeyInfo // import { KeyInfo } from 'auth'?
}
```
for `authService.introspect` return type.
Then use it to construct the `new Grant` in the auth middleware.

Reply:
I had to do 
```
class TokenInfo extends Grant {
    public readonly key: KeyInfo
    ...
}
```

because `Grant` is a class